### PR TITLE
IR-1282-Track-code-generated-assets-in-the-resource-manager

### LINKED ISSUE #9941: Integrate performance profiling and system profiling into tunable parameters
- [ ] Build ee-development-test-suite benchmark scenes & benchmark tests for:
  - [ ] Physics
  - [ ] Avatar rendering
  - [ ] IK calculations
  - [ ] ... figure out more hot spots
- [ ] Priority queues have dynamic limits based on performance profiler

### PR DESCRIPTION
**Pull request by MichaelEstes**
Jira linked copy of https://github.com/EtherealEngine/etherealengine/pull/9956

## Summary

#9941 
- Exposes engine version in vite.config
- Add functions for aggregating system performance
- Store GPU and device information in PerformanceState
- Pull request [82](https://github.com/EtherealEngine/ee-development-test-suite/pull/82) in ee-development-test-suite relies on these changes to run benchmarks

#9942 
- Adds GPU timing code through WebGL extensions with fallback for non-supported browsers in PerformanceState.ts and WebGLRendererSystem.ts

#9900 
- Adds hooks in resourceHooks.ts for managing code generated resources that allocate GPU memory through ResourceState.ts
- Splits code from components in the happy path that are creating code generated resources into their own components that use the managed hook flow

Misc
- Fix bugs around layers being used as layer masks
- Component tests
- Fixed small bug in the system debug window that was preventing system's frame time from being displayed

## References
closes #9941 #9900 #9942 

## QA Steps

### CODE REVIEW

File: .vscode/snippets.code-snippets [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+	"Define Component": {
```
Comment (DanielBelmes) [inline .vscode/snippets.code-snippets:2]: Ooooo this is nice. Might benefit from more in the future.

File: packages/client-core/i18n/en/common.json [hunk 2]
```diff
@@ -76,6 +76,7 @@
     "tick": "Tick",
     "state": "State",
     "systems": "Systems",
+    "uuid": "Avg Duration: ",
```
Comment (DanielBelmes) [inline packages/client-core/i18n/en/common.json]: doesn't look like the right string for this but I could be wrong.

Comment (MichaelEstes) [reply packages/client-core/i18n/en/common.json]: Ya, that's not very clear. I renamed it

**Review by DanielBelmes — APPROVED** [review]
Looking good. it is a lot so lets get another review. I QA'd and didn't find anything but there are issues with dev right now that need fixing that I found. Specifically around deleting a modelCompoent and adding that model back.

**Pull request merged**